### PR TITLE
Refine textclassifier with data without headers and refactor to temporal convolution 

### PR DESCRIPTION
**Pull request by psyyz10**
## What changes were proposed in this pull request?

Refine textclassifier with data without headers and refactor to temporal convolution

### CODE REVIEW

File: pyspark/bigdl/dataset/news20.py [hunk 1]
```diff
@@ -19,17 +19,17 @@
 import os
 import sys
 
-NEWS20_URL = 'http://qwone.com/~jason/20Newsgroups/20news-19997.tar.gz'  # noqa
+NEWS20_URL = 'http://qwone.com/~jason/20Newsgroups/20news-18828.tar.gz'  # noqa
```
Comment (zhichao-li) [inline pyspark/bigdl/dataset/news20.py]: so 18828 is the datasource without header and label?

File: pyspark/bigdl/nn/layer.py [hunk 2]
```diff
@@ -1079,7 +1079,7 @@ class TemporalMaxPooling(Layer):
 
     def __init__(self,
                  k_w,
-                 d_w,
+                 d_w=-1,
```
Comment (zhichao-li) [inline pyspark/bigdl/nn/layer.py]: add docs for the meaning of -1?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/NNPrimitive.scala [hunk 3]
```diff
@@ -1006,9 +1006,9 @@ private[nn] object NNPrimitive {
 
       var y = 0
       while (y < frameSize) {
-        val maxIndex = xp + y
+        val maxIndex = indices(xp + y).toInt - 1
```
Comment (zhichao-li) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/NNPrimitive.scala]: why changing this? do we need to add unitest to cover it?

File: pyspark/bigdl/dataset/news20.py (hunk 1 above)
Comment (psyyz10) [reply pyspark/bigdl/dataset/news20.py]: Yes

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/NNPrimitive.scala (hunk 3 above)
Comment (psyyz10) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/NNPrimitive.scala]: We have unit test, this just a bug, I fixed it, keep float same as double

Comment (zhichao-li) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/NNPrimitive.scala]: ok. just wondering why the existing unittest didn't capture this bug.

**Comment by zhichao-li** [comment]
LGTM

**Comment by zhichao-li** [comment]
btw, do we need to update the docs accordingly? i.e: update the accuracy

**Comment by zhichao-li** [comment]
any update for this?

**Comment by zhichao-li** [comment]
LGTM pending on validation

**Pull request merged**